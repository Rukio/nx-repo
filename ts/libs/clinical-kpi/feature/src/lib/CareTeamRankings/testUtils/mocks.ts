import {
  AuthenticatedUser,
  MarketMetrics,
  mockedAuthenticatedUser,
} from '@*company-data-covered*/clinical-kpi/data-access';
import {
  buildProvider,
  generateIncrementalMockedProviderMetrics,
} from '../../util/testUtils/builders/providerProfile';

export const mockedProviderMetricsResponse: MarketMetrics = {
  providerMetrics: [
    {
      provider: buildProvider(mockedAuthenticatedUser.id),
      metrics: generateIncrementalMockedProviderMetrics(19),
    },
  ],
};

export const mockAuthenticatedUserId: AuthenticatedUser['id'] =
  mockedProviderMetricsResponse.providerMetrics[0].provider.id.toString();
