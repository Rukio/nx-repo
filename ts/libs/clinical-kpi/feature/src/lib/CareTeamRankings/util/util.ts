import { LeaderHubMetricsData } from '@*company-data-covered*/clinical-kpi/data-access';
import {
  LeaderHubMetricsChangeKeys,
  LeaderHubMetricsKeys,
  LeaderHubMetricsRankKeys,
} from '../../constants';
import {
  getNumericMetricValue,
  getValidMetricValue,
} from '../../util/metricUtils';
import { Metrics } from '@*company-data-covered*/clinical-kpi/ui';

export type LeaderHubRankTableRow = {
  id: number;
  rank: number;
  name: string;
  value?: number;
  valueChange: number;
  position?: string;
};

export type BuildLeaderHubProvidersRankingParams = {
  metrics?: LeaderHubMetricsData[];
  key: LeaderHubMetricsKeys;
  changeKey: LeaderHubMetricsChangeKeys;
  type: Metrics;
  rankKey: LeaderHubMetricsRankKeys;
};

type LeaderHubRankingsData = {
  rankTableProviders: LeaderHubRankTableRow[];
};

const formatLeaderHubProviderData = (
  metric: LeaderHubMetricsData,
  key: LeaderHubMetricsKeys,
  changeKey: LeaderHubMetricsChangeKeys,
  rankKey: LeaderHubMetricsRankKeys,
  type: Metrics
): LeaderHubRankTableRow | null => {
  const value = metric[key];
  const valueChange = metric[changeKey];
  const provider = metric.provider;
  const rank = metric[rankKey];

  return {
    id: Number(metric.providerId),
    name: `${provider.firstName} ${provider.lastName}`,
    rank: parseInt(rank, 10),
    value: getValidMetricValue(type, value),
    valueChange: getNumericMetricValue(valueChange, type),
    position: provider.profile.position,
  };
};

export const buildProvidersLeaderHubRanking = ({
  metrics: marketMetrics,
  key,
  changeKey,
  type,
  rankKey,
}: BuildLeaderHubProvidersRankingParams): LeaderHubRankingsData => {
  if (!marketMetrics) {
    return {
      rankTableProviders: [],
    };
  }

  return {
    rankTableProviders: marketMetrics
      .map((metric) =>
        formatLeaderHubProviderData(metric, key, changeKey, rankKey, type)
      )
      .filter(
        (formattedProvider): formattedProvider is LeaderHubRankTableRow =>
          formattedProvider !== null
      ),
  };
};
