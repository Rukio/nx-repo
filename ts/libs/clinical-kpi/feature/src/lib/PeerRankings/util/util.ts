import {
  Market,
  AuthenticatedUser,
  MarketMetrics,
  ProviderMetrics,
  Provider as ProviderData,
} from '@*company-data-covered*/clinical-kpi/data-access';
import { Metrics } from '@*company-data-covered*/clinical-kpi/ui';
import {
  MetricsKeys,
  MetricsChangeKeys,
  ProviderPosition,
} from '../../constants';
import { Provider, RankingsData, SortOrder } from '../types';
import {
  formatProviderPosition,
  getNumericMetricValue,
  isValidMetricValue,
} from '../../util/metricUtils';

type BuildProvidersRankingParams = {
  metrics: MarketMetrics | undefined;
  key: MetricsKeys;
  changeKey: MetricsChangeKeys;
  sortOrder: SortOrder;
  authenticatedUserId: AuthenticatedUser['id'];
  includedProviderPositions?: ProviderPosition[];
  type: Metrics;
};

type FormatProviderParams = {
  provider: ProviderData;
  metrics: ProviderMetrics;
  key: MetricsKeys;
  changeKey: MetricsChangeKeys;
  includedProviderPositions?: ProviderPosition[];
  type: Metrics;
};

export type LeaderHubRankTable = {
  id: number;
  rank: number;
  name: string;
  value: number;
  valueChange: number;
  position?: string;
};

type FormattedProvider = Omit<Provider, 'rank'>;

type GroupedByValuesProviders = FormattedProvider[];

type ProviderValue = number;

export const isValidProviderPosition = (
  position: string
): position is ProviderPosition => {
  switch (position) {
    case ProviderPosition.APP:
    case ProviderPosition.DHMT:
      return true;
    default:
      return false;
  }
};

export const getFormattedProviderData = ({
  provider,
  metrics,
  key,
  changeKey,
  includedProviderPositions,
  type,
}: FormatProviderParams): FormattedProvider | null => {
  const { position } = provider.profile;
  const value = metrics[key];
  const valueChange = metrics[changeKey];

  const isPositionIncluded =
    includedProviderPositions &&
    includedProviderPositions.length !== 0 &&
    isValidProviderPosition(position)
      ? includedProviderPositions.includes(position)
      : true;

  if (
    metrics &&
    isValidMetricValue(value) &&
    isValidMetricValue(valueChange) &&
    isPositionIncluded
  ) {
    return {
      id: provider.id,
      name: `${provider.firstName} ${provider.lastName}`,
      value: getNumericMetricValue(value, type),
      valueChange: getNumericMetricValue(valueChange, type),
      avatarUrl: provider.avatarUrl ?? '',
      position: formatProviderPosition(provider.profile?.position),
    };
  }

  return null;
};

// TODO: [PT-1162] refactor PeerRankings feature by moving transformation function to the feature slice as selector
export const buildProvidersRanking = ({
  metrics: marketMetrics,
  key,
  changeKey,
  sortOrder,
  authenticatedUserId,
  includedProviderPositions,
  type,
}: BuildProvidersRankingParams): RankingsData => {
  if (!marketMetrics?.providerMetrics) {
    return {
      marqueeLeaderProviders: {
        firstPosition: [],
        secondPosition: [],
        thirdPosition: [],
      },
      rankTableProviders: [],
    };
  }
  const groupedProvidersData = marketMetrics.providerMetrics
    .reduce<[ProviderValue, GroupedByValuesProviders][]>(
      (acc, providerMetric) => {
        const formattedProvider = getFormattedProviderData({
          provider: providerMetric.provider,
          metrics: providerMetric.metrics,
          key: key,
          changeKey: changeKey,
          includedProviderPositions,
          type,
        });

        if (!formattedProvider) {
          return acc;
        }

        const existingProviderIndex = acc.findIndex(
          ([key]) => key === formattedProvider.value
        );

        if (existingProviderIndex !== -1) {
          acc[existingProviderIndex][1].push(formattedProvider);
        } else {
          acc.push([formattedProvider.value, [formattedProvider]]);
        }

        return acc;
      },
      []
    )
    .sort(([providerValueA], [providerValueB]) => {
      if (sortOrder === SortOrder.ASC) {
        return providerValueA - providerValueB;
      }

      return providerValueB - providerValueA;
    })
    .map<[number, Provider[]]>(([providerValue, groupedProviders], index) => {
      const groupedProvidersWithRank = groupedProviders.map<Provider>(
        (provider) => ({
          ...provider,
          rank: index + 1,
        })
      );

      return [providerValue, groupedProvidersWithRank];
    });

  const groupedProvidersArray = groupedProvidersData.map(
    ([, groupedProviders]) => groupedProviders
  );

  const [firstPosition, secondPosition, thirdPosition, ...tableElements] =
    groupedProvidersArray;

  const currentProviderData = groupedProvidersArray
    .flat()
    .find((row) => row.id.toString() === authenticatedUserId);

  return {
    currentProviderData,
    marqueeLeaderProviders: {
      firstPosition: firstPosition || [],
      secondPosition: secondPosition || [],
      thirdPosition: thirdPosition || [],
    },
    rankTableProviders: tableElements.flat().map((row) => ({
      ...row,
      name: row.id.toString() === authenticatedUserId ? row.name : '',
    })),
  };
};

export const getDefaultSelectedMarket = (
  markets: Market[]
): string | undefined => {
  if (!markets.length) {
    return undefined;
  }

  return markets[0].id;
};
