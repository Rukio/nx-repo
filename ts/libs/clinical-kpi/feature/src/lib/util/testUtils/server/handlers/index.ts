export * from './user';
export * from './metrics';
export * from './providerOverallMetrics';
export * from './providerMarkets';
export * from './marketProviderMetrics';
export * from './providerVisits';
