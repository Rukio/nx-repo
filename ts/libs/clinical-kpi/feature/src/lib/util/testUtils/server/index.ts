import { setupServer } from 'msw/node';
import {
  metricsHandlers,
  userHandlers,
  marketProvidersMetricsHandlers,
  providerOverallMetrics,
  providerMarkets,
  providerVisits,
} from './handlers';
const server = setupServer(
  ...[
    ...userHandlers,
    ...metricsHandlers,
    ...marketProvidersMetricsHandlers,
    ...providerOverallMetrics,
    ...providerMarkets,
    ...providerVisits,
  ]
);

export { server };
