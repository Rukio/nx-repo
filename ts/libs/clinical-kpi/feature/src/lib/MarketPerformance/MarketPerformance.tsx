import statsig from 'statsig-js';
import { skipToken } from '@reduxjs/toolkit/dist/query';
import { makeSxStyles, Grid } from '@*company-data-covered*/design-system';
import { IndividualMetric, Metrics } from '@*company-data-covered*/clinical-kpi/ui';
import {
  selectSelectedMarketId,
  useGetMarketLeaderHubMetricsQuery,
} from '@*company-data-covered*/clinical-kpi/data-access';
import {
  STATSIG_KEYS,
  LeaderHubMetricsChangeKeys,
  LeaderHubMetricsKeys,
  MetricsGoalKeys,
} from '../constants';

import PerformanceContainer from '../PerformanceContainer';
import { getNumericMetricValue } from '../util/metricUtils';
import { useSelector } from 'react-redux';

interface LeaderHubMarketMetricConfiguration {
  type: Metrics;
  key: Exclude<LeaderHubMetricsKeys, LeaderHubMetricsKeys.OnTaskPercent>;
  changeKey: Exclude<
    LeaderHubMetricsChangeKeys,
    LeaderHubMetricsChangeKeys.OnTaskPercent
  >;
}

const onSceneTimeConfiguration: LeaderHubMarketMetricConfiguration = {
  type: Metrics.OnSceneTime,
  key: LeaderHubMetricsKeys.OnSceneTime,
  changeKey: LeaderHubMetricsChangeKeys.OnSceneTime,
};
const npsConfiguration: LeaderHubMarketMetricConfiguration = {
  type: Metrics.NPS,
  key: LeaderHubMetricsKeys.NPS,
  changeKey: LeaderHubMetricsChangeKeys.NPS,
};

const surveyCaptureConfiguration: LeaderHubMarketMetricConfiguration = {
  type: Metrics.SurveyCapture,
  key: LeaderHubMetricsKeys.SurveyCapture,
  changeKey: LeaderHubMetricsChangeKeys.SurveyCapture,
};

const chartClosureConfiguration: LeaderHubMarketMetricConfiguration = {
  type: Metrics.ChartClosure,
  key: LeaderHubMetricsKeys.ChartClosure,
  changeKey: LeaderHubMetricsChangeKeys.ChartClosure,
};

const marketConfiguration: LeaderHubMarketMetricConfiguration[] = [
  onSceneTimeConfiguration,
  surveyCaptureConfiguration,
  chartClosureConfiguration,
  npsConfiguration,
];

const makeStyles = () =>
  makeSxStyles({
    metricContainer: {
      width: '100%',
    },
  });

const MarketPerformance = () => {
  const styles = makeStyles();
  const { selectedMarketId } = useSelector(selectSelectedMarketId);

  const {
    data: metrics,
    isLoading,
    isError,
  } = useGetMarketLeaderHubMetricsQuery(selectedMarketId ?? skipToken);

  const { value: metricsGoals } = statsig.getConfig(
    STATSIG_KEYS.DYNAMIC_CONFIGS.METRICS
  );

  return (
    <PerformanceContainer
      title="Market Performance"
      isLoading={isLoading}
      isError={isError || !metrics}
    >
      {metrics &&
        marketConfiguration.map((metric, index) => (
          <Grid
            item
            sm={3}
            sx={styles.metricContainer}
            key={`${metric.type}-${index}`}
          >
            <IndividualMetric
              type={metric.type}
              value={getNumericMetricValue(metrics[metric.key], metric.type)}
              valueChange={getNumericMetricValue(
                metrics[metric.changeKey],
                metric.type
              )}
              goal={metricsGoals[MetricsGoalKeys[metric.type]]}
              testIdPrefix={metric.type}
              possessiveOfMetric={"the Market's"}
            />
          </Grid>
        ))}
    </PerformanceContainer>
  );
};

export default MarketPerformance;
