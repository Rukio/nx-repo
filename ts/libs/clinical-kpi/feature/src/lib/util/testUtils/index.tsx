import { DefaultBodyType, DelayMode, rest } from 'msw';
import {
  render as rtlRender,
  RenderOptions as RTLRenderOptions,
} from '@testing-library/react';
import { ReactNode } from 'react';
import { Provider } from 'react-redux';
import { configureStore } from '@reduxjs/toolkit';
import { setupListeners } from '@reduxjs/toolkit/dist/query';
import userEvent from '@testing-library/user-event';
import { BrowserRouter } from 'react-router-dom';
import {
  RootState,
  store as clinicalKpiStore,
  AUTH_FEATURE_KEY,
  authSlice,
  CLINICAL_KPI_API_SLICE_KEY,
  clinicalKpiApiSlice,
  PEER_RANKINGS_KEY,
  peerRankingsSlice,
  PROVIDER_VISITS_TABLE_KEY,
  providerVisitsTableSlice,
  CARE_TEAM_RANKINGS_KEY,
  careTeamRankingsSlice,
  PROVIDER_SHIFTS_KEY,
  providerShiftsTableSlice,
  INDIVIDUAL_PERFORMANCE_POSITION_KEY,
  individualPerformancePositionSlice,
} from '@*company-data-covered*/clinical-kpi/data-access';
import UserProvider from '../../UserProvider';
import { server } from './server';

type RenderOptions = RTLRenderOptions & {
  preloadedState?: Partial<RootState>;
  store?: typeof clinicalKpiStore;
};

export function render(
  ui: React.ReactElement,
  {
    preloadedState,
    store = configureStore({
      reducer: {
        [AUTH_FEATURE_KEY]: authSlice.reducer,
        [CLINICAL_KPI_API_SLICE_KEY]: clinicalKpiApiSlice.reducer,
        [PEER_RANKINGS_KEY]: peerRankingsSlice.reducer,
        [PROVIDER_VISITS_TABLE_KEY]: providerVisitsTableSlice.reducer,
        [CARE_TEAM_RANKINGS_KEY]: careTeamRankingsSlice.reducer,
        [PROVIDER_SHIFTS_KEY]: providerShiftsTableSlice.reducer,
        [INDIVIDUAL_PERFORMANCE_POSITION_KEY]:
          individualPerformancePositionSlice.reducer,
      },
      preloadedState,
      middleware: (getDefaultMiddleware) =>
        getDefaultMiddleware({
          immutableCheck: false,
          serializableCheck: false,
        }).concat(clinicalKpiApiSlice.middleware),
    }),
    ...renderOptions
  }: RenderOptions = {}
) {
  setupListeners(store.dispatch);
  const Wrapper: React.FC<{ children: ReactNode }> = ({ children }) => {
    return (
      <BrowserRouter>
        <Provider store={store}>{children}</Provider>
      </BrowserRouter>
    );
  };

  return {
    user: userEvent.setup({ delay: null }),
    ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }),
    store,
  };
}

export const renderWithUserProvider = (
  component?: React.ReactNode,
  renderOptions?: RenderOptions
) => {
  return {
    ...render(<UserProvider>{component}</UserProvider>, renderOptions),
  };
};

type InterceptQueryProps = {
  url: string;
  data?: DefaultBodyType;
  delay?: DelayMode | number;
  statusCode?: number;
};
// TODO: [PT-1289] find a way to make mock intercept builders type-safe for each endpoint
export const interceptQuery = ({
  url,
  data,
  delay,
  statusCode,
}: InterceptQueryProps) => {
  server.use(
    rest.get(url, (_req, res, ctx) =>
      res.once(ctx.status(statusCode || 200), ctx.delay(delay), ctx.json(data))
    )
  );
};

export * from '@testing-library/react';
