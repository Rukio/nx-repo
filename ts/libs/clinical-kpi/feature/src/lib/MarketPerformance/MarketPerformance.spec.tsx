import { screen, waitFor } from '@testing-library/react';
import {
  DEFAULT_ERROR_ALERT_TEXT,
  Metrics,
  TEST_IDS,
} from '@*company-data-covered*/clinical-kpi/ui';
import {
  ProviderMetrics,
  RootState,
} from '@*company-data-covered*/clinical-kpi/data-access';
import { mockedProviderMetrics } from '@*company-data-covered*/clinical-kpi/data-access';
import { interceptQuery, renderWithUserProvider } from '../util/testUtils';
import {
  PERFORMANCE_TEST_IDS,
  mockedStatsigConfiguration,
} from '../PerformanceContainer';
import { MARKET_OVERALL_METRICS_INTERCEPT_URL } from '../util/testUtils/server/handlers';
import MarketPerformance from './MarketPerformance';

const PRELOADED_STATE: Partial<RootState> = {
  peerRankings: {
    selectedMarketId: '200',
  },
};

jest.mock('statsig-js', () => ({
  getConfig: jest.fn(() => mockedStatsigConfiguration),
}));

describe('MarketPerformance', () => {
  it('should render correctly when data is present', async () => {
    const { asFragment } = renderWithUserProvider(<MarketPerformance />, {
      preloadedState: PRELOADED_STATE,
    });

    await waitFor(() => {
      expect(
        screen.getByTestId(
          `${PERFORMANCE_TEST_IDS.SECTION}-${TEST_IDS.METRICS_SECTION.TITLE}`
        )
      ).toHaveTextContent('Market Performance');
    });
    expect(asFragment()).toMatchSnapshot();
  });

  it('should render correctly when data is present and some of the values are null', async () => {
    const mockData: ProviderMetrics = {
      ...mockedProviderMetrics,
      chartClosureRate: null,
      chartClosureRateChange: null,
    };
    interceptQuery({
      url: MARKET_OVERALL_METRICS_INTERCEPT_URL,
      data: { marketMetrics: mockData },
    });
    renderWithUserProvider(<MarketPerformance />, {
      preloadedState: PRELOADED_STATE,
    });
    const value = await screen.findByTestId(
      TEST_IDS.INDIVIDUAL_METRIC.VALUE(Metrics.ChartClosure)
    );

    expect(value.textContent).toBe('0%');

    const valueChange = await screen.findByTestId(
      TEST_IDS.INDIVIDUAL_METRIC.VALUE_CHANGE(Metrics.ChartClosure)
    );
    expect(valueChange.textContent).toBe('no change');
  });

  it('should render correct alert when metrics are empty', async () => {
    interceptQuery({
      url: MARKET_OVERALL_METRICS_INTERCEPT_URL,
      data: { marketMetrics: null },
    });
    renderWithUserProvider(<MarketPerformance />);
    const alert = await screen.findByText(DEFAULT_ERROR_ALERT_TEXT);

    expect(alert).toBeVisible();
  });

  it('should render placeholder when is loading', async () => {
    interceptQuery({
      url: MARKET_OVERALL_METRICS_INTERCEPT_URL,
      delay: 500,
    });
    renderWithUserProvider(<MarketPerformance />, {
      preloadedState: PRELOADED_STATE,
    });
    const placeholder = await screen.findByTestId(
      PERFORMANCE_TEST_IDS.PLACEHOLDER
    );

    expect(placeholder).toBeVisible();
  });
});
