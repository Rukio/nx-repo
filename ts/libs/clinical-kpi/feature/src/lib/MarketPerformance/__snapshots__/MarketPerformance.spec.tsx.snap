// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarketPerformance should render correctly when data is present 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 100%;
  margin-left: auto;
  box-sizing: border-box;
  margin-right: auto;
  display: block;
  padding-left: 16px;
  padding-right: 16px;
}

@media (min-width:600px) {
  .emotion-0 {
    padding-left: 24px;
    padding-right: 24px;
  }
}

@media (min-width:1200px) {
  .emotion-0 {
    max-width: 1200px;
  }
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  border-bottom: 2px solid;
  border-bottom-color: rgba(0, 0, 0, 0.12);
  border-bottom-style: solid;
  padding: 4px 0px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-2 {
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  width: 1em;
  height: 1em;
  display: inline-block;
  fill: currentColor;
  -webkit-flex-shrink: 0;
  -ms-flex-negative: 0;
  flex-shrink: 0;
  -webkit-transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: fill 200ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  font-size: 1.5rem;
  color: #1976d2;
}

.emotion-3 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 500;
  font-size: 1.25rem;
  line-height: 1.6;
  letter-spacing: 0.0075em;
  color: #1976d2;
  margin-left: 12px;
}

.emotion-4 {
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-flex-wrap: wrap;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  width: 100%;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  padding-top: 24px;
  padding-bottom: 8px;
}

@media (min-width:0px) {
  .emotion-4 {
    margin-top: -16px;
  }

  .emotion-4>.MuiGrid-item {
    padding-top: 16px;
  }
}

@media (min-width:600px) {
  .emotion-4 {
    margin-top: 0;
  }

  .emotion-4>.MuiGrid-item {
    padding-top: 0;
  }
}

@media (min-width:600px) {
  .emotion-4 {
    width: calc(100% + 16px);
    margin-left: -16px;
  }

  .emotion-4>.MuiGrid-item {
    padding-left: 16px;
  }
}

.emotion-5 {
  box-sizing: border-box;
  margin: 0;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
}

@media (min-width:600px) {
  .emotion-5 {
    -webkit-flex-basis: 25%;
    -ms-flex-preferred-size: 25%;
    flex-basis: 25%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 25%;
  }
}

@media (min-width:900px) {
  .emotion-5 {
    -webkit-flex-basis: 25%;
    -ms-flex-preferred-size: 25%;
    flex-basis: 25%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 25%;
  }
}

@media (min-width:1200px) {
  .emotion-5 {
    -webkit-flex-basis: 25%;
    -ms-flex-preferred-size: 25%;
    flex-basis: 25%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 25%;
  }
}

@media (min-width:1536px) {
  .emotion-5 {
    -webkit-flex-basis: 25%;
    -ms-flex-preferred-size: 25%;
    flex-basis: 25%;
    -webkit-box-flex: 0;
    -webkit-flex-grow: 0;
    -ms-flex-positive: 0;
    flex-grow: 0;
    max-width: 25%;
  }
}

.emotion-6 {
  background-color: #fff;
  color: rgba(0, 0, 0, 0.87);
  -webkit-transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: box-shadow 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-radius: 4px;
  box-shadow: 0px 2px 1px -1px rgba(0,0,0,0.2),0px 1px 1px 0px rgba(0,0,0,0.14),0px 1px 3px 0px rgba(0,0,0,0.12);
  overflow: hidden;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: left;
  -webkit-box-align: left;
  -ms-flex-align: left;
  align-items: left;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 100%;
  width: 100%;
  border-radius: 8px;
  padding: 16px;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 8px;
}

.emotion-8 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1rem;
  line-height: 1.75;
  letter-spacing: 0.00938em;
}

.emotion-10 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 2.125rem;
  line-height: 1.235;
  letter-spacing: 0.00735em;
  font-size: 1.5rem;
}

.emotion-11 {
  margin: 0;
  font-family: Open Sans;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: 8px;
  margin-right: 8px;
  color: #d32f2f;
}

.emotion-12 {
  margin: 0;
  color: #616161;
  padding: 8px;
}

.emotion-13 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 0.875rem;
  line-height: 1.43;
  letter-spacing: 0.01071em;
  color: #616161;
  padding: 8px;
}

.emotion-29 {
  margin: 0;
  font-family: Open Sans;
  margin-top: auto;
  margin-bottom: auto;
  margin-left: 8px;
  margin-right: 8px;
  color: #2e7d32;
}

<div
    class="MuiContainer-root MuiContainer-maxWidthLg emotion-0"
  >
    <div
      class="MuiBox-root emotion-1"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-colorPrimary MuiSvgIcon-fontSizeMedium emotion-2"
        data-testid="PersonIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
        />
      </svg>
      <h6
        class="MuiTypography-root MuiTypography-h6 emotion-3"
        data-testid="performance-section-metrics-section-title"
      >
        Market Performance
      </h6>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-sm-2 emotion-4"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-3 emotion-5"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-6"
        >
          <div
            class="MuiBox-root emotion-7"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 emotion-8"
              data-testid="OnSceneTime-individual-metric-type"
            >
              On Scene Time
            </h6>
          </div>
          <div
            class="MuiBox-root emotion-7"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 emotion-10"
              data-testid="OnSceneTime-individual-metric-value"
            >
              97.26 mins
            </h4>
            <span
              class="MuiTypography-root MuiTypography-label emotion-11"
              data-testid="OnSceneTime-individual-metric-change"
            >
              +290.52 mins
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-label emotion-12"
            data-testid="OnSceneTime-individual-metric-performance"
          >
            Since last week, the Market's On Scene Time went up 290.52 mins
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-13"
            data-testid="OnSceneTime-individual-metric-goal"
          >
            Goal: 30 mins
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-3 emotion-5"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-6"
        >
          <div
            class="MuiBox-root emotion-7"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 emotion-8"
              data-testid="SurveyCapture-individual-metric-type"
            >
              Survey Capture
            </h6>
          </div>
          <div
            class="MuiBox-root emotion-7"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 emotion-10"
              data-testid="SurveyCapture-individual-metric-value"
            >
              91.84%
            </h4>
            <span
              class="MuiTypography-root MuiTypography-label emotion-11"
              data-testid="SurveyCapture-individual-metric-change"
            >
              -2.59%
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-label emotion-12"
            data-testid="SurveyCapture-individual-metric-performance"
          >
            Since last week, the Market's Survey Capture went down 2.59%
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-13"
            data-testid="SurveyCapture-individual-metric-goal"
          >
            Goal: 80%
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-3 emotion-5"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-6"
        >
          <div
            class="MuiBox-root emotion-7"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 emotion-8"
              data-testid="ChartClosure-individual-metric-type"
            >
              Chart Closure
            </h6>
          </div>
          <div
            class="MuiBox-root emotion-7"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 emotion-10"
              data-testid="ChartClosure-individual-metric-value"
            >
              73.39%
            </h4>
            <span
              class="MuiTypography-root MuiTypography-label emotion-29"
              data-testid="ChartClosure-individual-metric-change"
            >
              +4.41%
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-label emotion-12"
            data-testid="ChartClosure-individual-metric-performance"
          >
            Since last week, the Market's Chart Closure went up 4.41%
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-13"
            data-testid="ChartClosure-individual-metric-goal"
          >
            Goal: 99%
          </p>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-3 emotion-5"
      >
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root emotion-6"
        >
          <div
            class="MuiBox-root emotion-7"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle1 emotion-8"
              data-testid="NPS-individual-metric-type"
            >
              NPS
            </h6>
          </div>
          <div
            class="MuiBox-root emotion-7"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 emotion-10"
              data-testid="NPS-individual-metric-value"
            >
              85.77
            </h4>
            <span
              class="MuiTypography-root MuiTypography-label emotion-11"
              data-testid="NPS-individual-metric-change"
            >
              -0.34
            </span>
          </div>
          <span
            class="MuiTypography-root MuiTypography-label emotion-12"
            data-testid="NPS-individual-metric-performance"
          >
            Since last week, the Market's NPS went down 0.34
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 emotion-13"
            data-testid="NPS-individual-metric-goal"
          >
            Goal: &gt;=95
          </p>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
